modules/src/ics03_connection/handler/verify.rs

```diff
@@ -96,7 +96,7 @@ pub fn verify_connection_proof(
             proof_height,
             connection_end.counterparty().prefix(),
             proof,
-            &connection_end.counterparty().connection_id().unwrap(),
+            connection_end.counterparty().connection_id(),
```

Comment:
If the connection is in the `Init` state, `connection_end.counterparty().connection_id()` is a `None`, and thus it's not okay to `unwrap`.